### PR TITLE
MultiButton

### PR DESCRIPTION
**Pull request by hborghols**
### This PR:

proposed release: `patch`

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable)

**Adds** ✨
- `MultiButton` component

### CODE REVIEW

File: src/components/MultiButton/style.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    &:active {
+        transform: translateY(2px);
+    }
+`;
+
+const StyledWindow = withProps<StyledMultiButtonPropsType, HTMLDivElement>(styled.div)`
+    box-sizing: border-box;
+    background: ${({ theme }): string => theme.MultiButton.window.backgroundColor}
+    overflow: hidden;
+    max-width: 350px;
```
Comment (CarloPalinckx) [inline src/components/MultiButton/style.tsx]: would be nice if this was dividable by 12. This way it possibly visually aligns with other elements. Maybe 360?

File: src/components/MultiButton/style.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        transform: translateY(0px);
+    }
+`;
+
+const StyledWrapper = withProps<StyledMultiButtonPropsType, HTMLDivElement>(styled.div)`
+    border-radius: ${({ theme }): string => theme.MultiButton.window.borderRadius}
+    box-shadow: ${({ isOpen, theme }): string => (isOpen ? theme.MultiButton.button.active.boxShadow : '')};
+    transform: translateZ(0) translate3d(0, 0, 0);
+    transition: transform 0.1s, background 0.3s, color 0.3s, box-shadow 0.1s, border 0.3s;
+    &:active {
```
Comment (CarloPalinckx) [inline src/components/MultiButton/style.tsx]: nit: newline

File: src/components/MultiButton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+type StateType = {
+    isOpen: boolean;
+    selectedOption: OptionBase;
+};
+
+class MultiButton<GenericOption extends OptionBase> extends Component<PropsType<GenericOption>, StateType> {
+    private windowRef: RefObject<HTMLDivElement>;
+    private buttonRef: RefObject<HTMLDivElement>;
+    private defaultOption: OptionBase = this.props.options.filter(option => {
```
Comment (CarloPalinckx) [inline src/components/MultiButton/index.tsx]: nit: newline

File: src/components/MultiButton/index.tsx [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    value: string;
+    label: string;
+    description: string;
+    default?: boolean;
+    action?(): void;
+};
+
+type PropsType<GenericOption extends OptionBase> = {
+    placement?: PlacementType;
+    options: Array<GenericOption>;
```
Comment (CarloPalinckx) [inline src/components/MultiButton/index.tsx]: Does this need to be generic? It seems like you always expect the same Option type.

**Comment by CarloPalinckx** [comment]
screen shot 2018-08-29 at 10 30 36
The chevron seems a bit skewed.

File: src/components/MultiButton/index.tsx [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+            () => {
+                if (this.state.selectedOption.action) {
+                    this.state.selectedOption.action();
+                }
+                this.close();
+            },
+        );
+    };
+
+    public renderButtons = (): JSX.Element => {
```
Comment (CarloPalinckx) [inline src/components/MultiButton/index.tsx]: this can be private

File: src/components/MultiButton/index.tsx [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+        return (
+            <>
+                <StyledMultiButton
```
Comment (CarloPalinckx) [inline src/components/MultiButton/index.tsx]: Maybe there is a way to make this more composable? You are pretty tied to the props this Multibutton exposes now . Couldn't you use the children as a way to inject an actual `<Button />`? This decouples the button part of this component.

File: src/components/MultiButton/style.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    border-bottom-right-radius: 0px;
+
+    &:active {
+        transform: translateY(0px);
+    }
+`;
+
+const StyledChevronButton = styled(Button)`
+    align-items: center;
+    border-left: solid 1px #dbdfe6;
```
Comment (woutsluiter) [inline src/components/MultiButton/style.tsx]: The color should be themable

**Comment by CarloPalinckx** [comment]
screen shot 2018-08-29 at 12 38 26
The focus state isn't applied to the chevron button.

File: src/components/MultiButton/index.tsx [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+                                                        <Box margin={trbl(0, 12, 0, 0)}>
+                                                            {option.value === this.state.selectedOption.value && (
+                                                                <Icon size="medium" icon="checkmark" />
+                                                            )}
+                                                            {option.value !== this.state.selectedOption.value && (
+                                                                <Box width={'18px'} />
+                                                            )}
+                                                        </Box>
+                                                        <div>
+                                                            <Text
```
[the remark this answers is not in the document]
Comment (hborghols) [reply src/components/MultiButton/index.tsx]: Ceci le design

File: src/themes/MosTheme/MosTheme.theme.ts [hunk 9]
```diff
@@ -479,6 +479,22 @@ const theme: ThemeType = {
...
+            secondaryColor: silver.base,
+            borderColor: silver.darker3,
+            borderRadius: roundness.base,
+            boxShadow: '0 2px 6px 0 rgba(0,0,0,0.10)',
+        },
+        button: {
+            active: {
+                boxShadow: 'inset 0 -2px 0 0 rgba(0,0,0,0.20), 0 0 0 2px rgba(136,151,157,0.20)',
+            },
+            dividerColor: silver.darker3,
```
Comment (woutsluiter) [inline src/themes/MosTheme/MosTheme.theme.ts]: Would be nice if this is themeable by Button variant

File: src/components/MultiButton/test.tsx [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+                default: true,
+                action: actionMock1,
+            },
+            {
+                value: 'Optionmock 2',
+                label: '',
+                description: '',
+                action: actionMock2,
+            },
+        ];
```
Comment (woutsluiter) [inline src/components/MultiButton/test.tsx]: Nit: newline

File: src/components/MultiButton/test.tsx [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+
+        const optionsMock = [
+            {
+                value: 'Optionmock',
+                label: 'Optionmock',
+                description: '',
+                action: actionMock1,
+            },
+        ];
+        const component = mountWithTheme(<MultiButton variant={'secondary'} options={optionsMock} />);
```
Comment (woutsluiter) [inline src/components/MultiButton/test.tsx]: newline

File: src/components/MultiButton/test.tsx [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+                action: actionMock1,
+            },
+            {
+                value: 'Optionmock 2',
+                label: '',
+                description: '',
+                action: actionMock2,
+            },
+        ];
+        const component = mountWithTheme(<MultiButton variant={'secondary'} options={optionsMock} />);
```
Comment (woutsluiter) [inline src/components/MultiButton/test.tsx]: newline

File: src/components/MultiButton/test.tsx [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+
+        expect(actionMock1).not.toHaveBeenCalled();
+        expect(actionMock2).toHaveBeenCalled();
+    });
+
+    it('should render with a custom distance and offset', () => {
+        const component = mountWithTheme(
+            <MultiButton variant={'secondary'} options={options} offset={20} distance={6} />,
+        );
+        const arrowButton = component.find(StyledChevronButton);
```
Comment (woutsluiter) [inline src/components/MultiButton/test.tsx]: Newline

**Pull request merged**